core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -1831,6 +1832,22 @@ class KafkaConfigTest {
     assertTrue(config.isNewGroupCoordinatorEnabled)
   }
 
+  @Test
+  def testGroupProtocolMigrationPolicy(): Unit = {
+    val props = new Properties()
+    props.putAll(kraftProps())
+
+    // Invalid GroupProtocolMigrationPolicy value.
+    props.put(KafkaConfig.ConsumerGroupMigrationPolicyProp, "foo")
+    assertThrows(classOf[ConfigException], () => KafkaConfig.fromProps(props))
+
+    ConsumerGroupMigrationPolicy.values().foreach { policy =>
+      props.put(KafkaConfig.ConsumerGroupMigrationPolicyProp, policy.toString)
```

Comment:
Is it case sensitive?

Reply:
Though it's not in `ConsumerGroupMigrationPolicy#parse`, in this test `KafkaConfig.fromProps` requires case sensitivity.